pyprep/prep_pipeline.py

```diff
@@ -133,14 +145,23 @@ def fit(self):
 
             # Step 2: Removing line noise
             linenoise = self.prep_params["line_freqs"]
-            self.EEG_clean = mne.filter.notch_filter(
-                self.EEG_new,
-                Fs=self.sfreq,
-                freqs=linenoise,
-                method="spectrum_fit",
-                mt_bandwidth=2,
-                p_value=0.01,
-            )
+            if self.filter_kwargs is None:
+                self.EEG_clean = mne.filter.notch_filter(
+                    self.EEG_new,
+                    Fs=self.sfreq,
+                    freqs=linenoise,
+                    method="spectrum_fit",
+                    mt_bandwidth=2,
+                    p_value=0.01,
+                    filter_length="10s",
+                )
+            else:
+                self.EEG_clean = mne.filter.notch_filter(
+                    self.EEG_new,
+                    Fs=self.sfreq,
+                    freqs=linenoise,
+                    **self.filter_kwargs,
+                )
```

Comment:
instead of this if-else, something more robust would be:

```Python
# set up our defaults ... this goes on line 123
self.filter_kwargs = {
    "x": self.EEG_new,
    "Fs": self.sfreq,
    "freqs": linenoise,
    "method": "spectrum_fit",
    "mt_bandwidth": 2,
    "p_value": 0.01,
    "filter_length": "10s",
}

# now update our defaults with the kwargs
self.filter_kwargs.update(filter_kwargs)

# finally, call mne.filter.notch_filter without if else
self.EEG_clean = mne.filter.notch_filter(**self.filter_kwargs)

```

what do you think @yjmantilla ?

Reply:
hmm Im worried about what happens when the user sets for example the method to 'fir' and is expecting the filter length to be the notch filter default 'auto' (so he doesn't set the filter length explicitly) but instead he gests a 10s filter length without him knowing.

Reply:
> Also some parameters aren't meaningful when one changes the filter method and supposedly they should be ignored but I guess that could potentially lead to some unknown bugs.

true, good point. We can leave it like you proposed then.